pkg/util/docker/uts.go

```diff
@@ -14,20 +14,34 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
 	"github.com/DataDog/datadog-agent/pkg/util/containers"
 	"github.com/DataDog/datadog-agent/pkg/util/containers/providers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/docker/docker/api/types/container"
 )
 
-// GetAgentContainerUTSMode provides the UTS mode of the Agent container
-// To get this info in an optimal way, consider calling util.GetAgentUTSMode instead to benefit from the cache
-func GetAgentContainerUTSMode(ctx context.Context) (containers.UTSMode, error) {
-	agentCID, _ := providers.ContainerImpl().GetAgentCID()
-	return GetContainerUTSMode(ctx, agentCID)
+// GetAgentUTSMode retrieves from Docker the UTS mode of the Agent container
+func GetAgentUTSMode(ctx context.Context) (containers.UTSMode, error) {
+	cacheUTSModeKey := cache.BuildAgentKey("utsMode")
+	if cacheUTSMode, found := cache.Cache.Get(cacheUTSModeKey); found {
+		return cacheUTSMode.(containers.UTSMode), nil
+	}
+
+	log.Debugf("GetAgentUTSMode trying docker")
+	utsMode, err := getContainerUTSMode(ctx)
+	cache.Cache.Set(cacheUTSModeKey, utsMode, cache.NoExpiration)
```

Comment:
I don't think the result from `getContainerUTSMode` is going to change during the lifetime of the Agent. So we can cache it to avoid checking every time.